Metrics extension causes issues with Resteasy in native runtime

**Describe the bug**
I followed the [Creating Your First Application](https://quarkus.io/guides/getting-started) guide and created a simple JAX-RS/Resteasy app. I have my code uploaded on [Github](https://github.com/saharsh-samples/sample-quarkus-app). The app runs fine doing a JVM based build. However if I do a native build using the [multi stage Dockerfile](https://quarkus.io/guides/building-native-image#creating-a-container-with-a-multi-stage-docker-build), I start getting MethodNotFoundException errors. After some debugging it looks like the issue is inclusion of the metrics extension. Creating the app as is with the exception of adding the metrics stuff causes the native app to behave just fine.

**Expected behavior**
Runtime of native app behaves the way runtime of the jvm app does.

**Actual behavior**
Any HTTP requests (including readiness health check) results in a MethodNotFoundException in the runtime of the native app.


**To Reproduce**
Steps to reproduce the behavior:
1. Clone [my repo](https://github.com/saharsh-samples/sample-quarkus-app)
2. Build native executable for the app (e.g. `./mvnw clean package -Pnative -Dquarkus.native.container-runtime=podman`)
3. Run the app (will run on port 8080) - `./target/sample-quarkus-app-1.0-SNAPSHOT-runner &`
4. Consume the app's endpoints:
    - curl localhost:8080/api/values (causes exception - see [relevant code](https://github.com/saharsh-samples/sample-quarkus-app/blob/master/src/main/java/org/saharsh/samples/quarkus/resources/ValuesResource.java))
    - curl localhost:8080/health/ready (causes exception - see [relevant code](https://github.com/saharsh-samples/sample-quarkus-app/blob/master/src/main/java/org/saharsh/samples/quarkus/health/ReadinessCheck.java))
    - curl curl localhost:8080/health/live (works OK - see [relevant code](https://github.com/saharsh-samples/sample-quarkus-app/blob/master/src/main/java/org/saharsh/samples/quarkus/health/LivenessCheck.java))

**Configuration**
```properties
# Add your application.properties here, if applicable.

```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: `Linux sahsingh 5.3.16-300.fc31.x86_64 #1 SMP Fri Dec 13 17:59:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`
 - Output of `java -version`: 
```
openjdk version "11" 2018-09-25
OpenJDK Runtime Environment 18.9 (build 11+28)
OpenJDK 64-Bit Server VM 18.9 (build 11+28, mixed mode)
```
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: `1.2.0.Final`

**Additional context**
(Add any other context about the problem here.)

Here's an example of the stack trace:

```
java.lang.NoSuchMethodException: org.saharsh.samples.quarkus.service.InMemoryStorageService.<init>
	at io.quarkus.arc.impl.Reflections.findConstructor(Reflections.java:46)
	at org.saharsh.samples.quarkus.service.InMemoryStorageService_Bean.create(InMemoryStorageService_Bean.zig:103)
	at org.saharsh.samples.quarkus.service.InMemoryStorageService_Bean.create(InMemoryStorageService_Bean.zig:191)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:80)
	at io.quarkus.arc.impl.ComputingCache$CacheFunction.lambda$apply$0(ComputingCache.java:99)
	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
	at io.quarkus.arc.impl.ComputingCache.getValue(ComputingCache.java:41)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:25)
	at org.saharsh.samples.quarkus.service.InMemoryStorageService_ClientProxy.arc$delegate(InMemoryStorageService_ClientProxy.zig:340)
	at org.saharsh.samples.quarkus.service.InMemoryStorageService_ClientProxy.isReady(InMemoryStorageService_ClientProxy.zig:173)
	at org.saharsh.samples.quarkus.health.ReadinessCheck.call(ReadinessCheck.java:26)
	at org.saharsh.samples.quarkus.health.ReadinessCheck_ClientProxy.call(ReadinessCheck_ClientProxy.zig:175)
	at io.smallrye.health.SmallRyeHealthReporter.jsonObject(SmallRyeHealthReporter.java:154)
	at io.smallrye.health.SmallRyeHealthReporter.fillCheck(SmallRyeHealthReporter.java:141)
	at io.smallrye.health.SmallRyeHealthReporter.processChecks(SmallRyeHealthReporter.java:129)
	at io.smallrye.health.SmallRyeHealthReporter.getHealth(SmallRyeHealthReporter.java:107)
	at io.smallrye.health.SmallRyeHealthReporter.getReadiness(SmallRyeHealthReporter.java:97)
	at io.smallrye.health.SmallRyeHealthReporter_ClientProxy.getReadiness(SmallRyeHealthReporter_ClientProxy.zig:101)
	at io.quarkus.smallrye.health.runtime.SmallRyeReadinessHandler.handle(SmallRyeReadinessHandler.java:40)
	at io.quarkus.smallrye.health.runtime.SmallRyeReadinessHandler.handle(SmallRyeReadinessHandler.java:32)
	at io.quarkus.vertx.http.runtime.ResumeHandler.handle(ResumeHandler.java:19)
	at io.quarkus.vertx.http.runtime.ResumeHandler.handle(ResumeHandler.java:6)
	at io.vertx.ext.web.impl.BlockingHandlerDecorator.lambda$handle$0(BlockingHandlerDecorator.java:48)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:316)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:460)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
Caused by: java.lang.NoSuchMethodException: org.saharsh.samples.quarkus.service.InMemoryStorageService.<init>
	at java.lang.Class.getConstructor0(DynamicHub.java:3082)
	at java.lang.Class.getDeclaredConstructor(DynamicHub.java:2178)
	at io.quarkus.arc.impl.Reflections.findConstructor(Reflections.java:44)
	... 30 more
```

--- a/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java
+++ b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java
@@ -164,7 +164,7 @@ public class BeanGenerator extends AbstractGenerator {
 
         MethodCreator constructor = initConstructor(classOutput, beanCreator, bean, baseName, Collections.emptyMap(),
                 Collections.emptyMap(),
-                annotationLiterals);
+                annotationLiterals, reflectionRegistration);
 
         FieldCreator params = beanCreator.getFieldCreator(FIELD_NAME_PARAMS, Map.class)
                 .setModifiers(ACC_PRIVATE | ACC_FINAL);
@@ -280,7 +280,7 @@ public class BeanGenerator extends AbstractGenerator {
         createProviderFields(beanCreator, bean, injectionPointToProviderSupplierField, interceptorToProviderSupplierField);
         createConstructor(classOutput, beanCreator, bean, baseName, injectionPointToProviderSupplierField,
                 interceptorToProviderSupplierField,
-                annotationLiterals);
+                annotationLiterals, reflectionRegistration);
 
         implementGetIdentifier(bean, beanCreator);
         implementSupplierGet(beanCreator);
@@ -375,7 +375,7 @@ public class BeanGenerator extends AbstractGenerator {
 
         createProviderFields(beanCreator, bean, injectionPointToProviderField, Collections.emptyMap());
         createConstructor(classOutput, beanCreator, bean, baseName, injectionPointToProviderField, Collections.emptyMap(),
-                annotationLiterals);
+                annotationLiterals, reflectionRegistration);
 
         implementGetIdentifier(bean, beanCreator);
         implementSupplierGet(beanCreator);
@@ -457,7 +457,7 @@ public class BeanGenerator extends AbstractGenerator {
 
         createProviderFields(beanCreator, bean, Collections.emptyMap(), Collections.emptyMap());
         createConstructor(classOutput, beanCreator, bean, baseName, Collections.emptyMap(), Collections.emptyMap(),
-                annotationLiterals);
+                annotationLiterals, reflectionRegistration);
 
         implementGetIdentifier(bean, beanCreator);
         implementSupplierGet(beanCreator);
@@ -530,16 +530,18 @@ public class BeanGenerator extends AbstractGenerator {
     protected void createConstructor(ClassOutput classOutput, ClassCreator beanCreator, BeanInfo bean, String baseName,
             Map<InjectionPointInfo, String> injectionPointToProviderField,
             Map<InterceptorInfo, String> interceptorToProviderField,
-            AnnotationLiteralProcessor annotationLiterals) {
+            AnnotationLiteralProcessor annotationLiterals,
+            ReflectionRegistration reflectionRegistration) {
         initConstructor(classOutput, beanCreator, bean, baseName, injectionPointToProviderField, interceptorToProviderField,
-                annotationLiterals)
+                annotationLiterals, reflectionRegistration)
                         .returnValue(null);
     }
 
     protected MethodCreator initConstructor(ClassOutput classOutput, ClassCreator beanCreator, BeanInfo bean, String baseName,
             Map<InjectionPointInfo, String> injectionPointToProviderField,
             Map<InterceptorInfo, String> interceptorToProviderField,
-            AnnotationLiteralProcessor annotationLiterals) {
+            AnnotationLiteralProcessor annotationLiterals,
+            ReflectionRegistration reflectionRegistration) {
 
         // First collect all param types
         List<String> parameterTypes = new ArrayList<>();
@@ -595,8 +597,8 @@ public class BeanGenerator extends AbstractGenerator {
             if (builtinBean != null) {
                 builtinBean.getGenerator()
                         .generate(new GeneratorContext(classOutput, bean.getDeployment(), injectionPoint, beanCreator,
-                                constructor,
-                                injectionPointToProviderField.get(injectionPoint), annotationLiterals, bean));
+                                constructor, injectionPointToProviderField.get(injectionPoint), annotationLiterals, bean,
+                                reflectionRegistration));
             } else {
 
                 if (BuiltinScope.DEPENDENT.is(injectionPoint.getResolvedBean().getScope()) && (injectionPoint.getResolvedBean()
@@ -606,7 +608,7 @@ public class BeanGenerator extends AbstractGenerator {
                     // Injection point resolves to a dependent bean that injects InjectionPoint metadata and so we need to wrap the injectable
                     // reference provider
                     ResultHandle wrapHandle = wrapCurrentInjectionPoint(classOutput, beanCreator, bean, constructor,
-                            injectionPoint, paramIdx++, tccl);
+                            injectionPoint, paramIdx++, tccl, reflectionRegistration);
                     ResultHandle wrapSupplierHandle = constructor.newInstance(
                             MethodDescriptors.FIXED_VALUE_SUPPLIER_CONSTRUCTOR, wrapHandle);
                     constructor.writeInstanceField(
@@ -946,6 +948,7 @@ public class BeanGenerator extends AbstractGenerator {
                     paramsHandles[1] = paramsArray;
                     constructorHandle = create.invokeStaticMethod(MethodDescriptors.REFLECTIONS_FIND_CONSTRUCTOR,
                             paramsHandles);
+                    reflectionRegistration.registerMethod(constructorInjection.get().target.asMethod());
                 } else {
                     // constructor = Reflections.findConstructor(Foo.class)
                     ResultHandle[] paramsHandles = new ResultHandle[2];
@@ -953,6 +956,8 @@ public class BeanGenerator extends AbstractGenerator {
                     paramsHandles[1] = create.newArray(Class.class, create.load(0));
                     constructorHandle = create.invokeStaticMethod(MethodDescriptors.REFLECTIONS_FIND_CONSTRUCTOR,
                             paramsHandles);
+                    MethodInfo noArgsConstructor = bean.getTarget().get().asClass().method(Methods.INIT);
+                    reflectionRegistration.registerMethod(noArgsConstructor);
                 }
 
                 List<ResultHandle> providerHandles = newProviderHandles(bean, beanCreator, create,
@@ -1587,13 +1592,14 @@ public class BeanGenerator extends AbstractGenerator {
     }
 
     private ResultHandle wrapCurrentInjectionPoint(ClassOutput classOutput, ClassCreator beanCreator, BeanInfo bean,
-            MethodCreator constructor, InjectionPointInfo injectionPoint, int paramIdx, ResultHandle tccl) {
+            MethodCreator constructor, InjectionPointInfo injectionPoint, int paramIdx, ResultHandle tccl,
+            ReflectionRegistration reflectionRegistration) {
         ResultHandle requiredQualifiersHandle = collectQualifiers(classOutput, beanCreator, bean.getDeployment(), constructor,
                 injectionPoint,
                 annotationLiterals);
         ResultHandle annotationsHandle = collectAnnotations(classOutput, beanCreator, bean.getDeployment(), constructor,
                 injectionPoint, annotationLiterals);
-        ResultHandle javaMemberHandle = getJavaMemberHandle(constructor, injectionPoint);
+        ResultHandle javaMemberHandle = getJavaMemberHandle(constructor, injectionPoint, reflectionRegistration);
 
         return constructor.newInstance(
                 MethodDescriptor.ofConstructor(CurrentInjectionPointProvider.class, InjectableBean.class,
@@ -1605,17 +1611,19 @@ public class BeanGenerator extends AbstractGenerator {
     }
 
     static ResultHandle getJavaMemberHandle(MethodCreator constructor,
-            InjectionPointInfo injectionPoint) {
+            InjectionPointInfo injectionPoint, ReflectionRegistration reflectionRegistration) {
         ResultHandle javaMemberHandle;
         if (Kind.FIELD.equals(injectionPoint.getTarget().kind())) {
             FieldInfo field = injectionPoint.getTarget().asField();
             javaMemberHandle = constructor.invokeStaticMethod(MethodDescriptors.REFLECTIONS_FIND_FIELD,
                     constructor.loadClass(field.declaringClass().name().toString()),
                     constructor.load(field.name()));
+            reflectionRegistration.registerField(field);
         } else {
             MethodInfo method = injectionPoint.getTarget().asMethod();
+            reflectionRegistration.registerMethod(method);
             if (method.name().equals(Methods.INIT)) {
-                // Reflections.findConstructor(org.foo.SimpleBean.class,java.lang.String.class)                
+                // Reflections.findConstructor(org.foo.SimpleBean.class,java.lang.String.class)
                 ResultHandle[] paramsHandles = new ResultHandle[2];
                 paramsHandles[0] = constructor.loadClass(method.declaringClass().name().toString());
                 ResultHandle paramsArray = constructor.newArray(Class.class, constructor.load(method.parameters().size()));

--- a/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BuiltinBean.java
+++ b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BuiltinBean.java
@@ -41,7 +41,8 @@ enum BuiltinBean {
         ResultHandle annotationsHandle = BeanGenerator.collectAnnotations(ctx.classOutput, ctx.clazzCreator, ctx.beanDeployment,
                 ctx.constructor,
                 ctx.injectionPoint, ctx.annotationLiterals);
-        ResultHandle javaMemberHandle = BeanGenerator.getJavaMemberHandle(ctx.constructor, ctx.injectionPoint);
+        ResultHandle javaMemberHandle = BeanGenerator.getJavaMemberHandle(ctx.constructor, ctx.injectionPoint,
+                ctx.reflectionRegistration);
         ResultHandle instanceProvider = ctx.constructor.newInstance(
                 MethodDescriptor.ofConstructor(InstanceProvider.class, java.lang.reflect.Type.class, Set.class,
                         InjectableBean.class, Set.class, Member.class, int.class),
@@ -238,10 +239,12 @@ enum BuiltinBean {
         final String providerName;
         final AnnotationLiteralProcessor annotationLiterals;
         final InjectionTargetInfo targetInfo;
+        final ReflectionRegistration reflectionRegistration;
 
         public GeneratorContext(ClassOutput classOutput, BeanDeployment beanDeployment, InjectionPointInfo injectionPoint,
                 ClassCreator clazzCreator, MethodCreator constructor, String providerName,
-                AnnotationLiteralProcessor annotationLiterals, InjectionTargetInfo targetInfo) {
+                AnnotationLiteralProcessor annotationLiterals, InjectionTargetInfo targetInfo,
+                ReflectionRegistration reflectionRegistration) {
             this.classOutput = classOutput;
             this.beanDeployment = beanDeployment;
             this.injectionPoint = injectionPoint;
@@ -250,6 +253,7 @@ enum BuiltinBean {
             this.providerName = providerName;
             this.annotationLiterals = annotationLiterals;
             this.targetInfo = targetInfo;
+            this.reflectionRegistration = reflectionRegistration;
         }
     }
 

--- a/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorGenerator.java
+++ b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorGenerator.java
@@ -89,8 +89,7 @@ public class InterceptorGenerator extends BeanGenerator {
 
         createProviderFields(interceptorCreator, interceptor, injectionPointToProviderField, interceptorToProviderField);
         createConstructor(classOutput, interceptorCreator, interceptor, baseName, injectionPointToProviderField,
-                interceptorToProviderField,
-                bindings.getFieldDescriptor());
+                interceptorToProviderField, bindings.getFieldDescriptor(), reflectionRegistration);
 
         implementGetIdentifier(interceptor, interceptorCreator);
         implementSupplierGet(interceptorCreator);
@@ -116,11 +115,11 @@ public class InterceptorGenerator extends BeanGenerator {
     protected void createConstructor(ClassOutput classOutput, ClassCreator creator, InterceptorInfo interceptor,
             String baseName,
             Map<InjectionPointInfo, String> injectionPointToProviderField,
-            Map<InterceptorInfo, String> interceptorToProviderField, FieldDescriptor bindings) {
+            Map<InterceptorInfo, String> interceptorToProviderField, FieldDescriptor bindings,
+            ReflectionRegistration reflectionRegistration) {
 
         MethodCreator constructor = initConstructor(classOutput, creator, interceptor, baseName, injectionPointToProviderField,
-                interceptorToProviderField,
-                annotationLiterals);
+                interceptorToProviderField, annotationLiterals, reflectionRegistration);
 
         // Bindings
         // bindings = new HashSet<>()

--- a/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ObserverGenerator.java
+++ b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ObserverGenerator.java
@@ -141,7 +141,7 @@ public class ObserverGenerator extends AbstractGenerator {
 
         createProviderFields(observerCreator, observer, injectionPointToProviderField);
         createConstructor(classOutput, observerCreator, observer, baseName.toString(), injectionPointToProviderField,
-                annotationLiterals);
+                annotationLiterals, reflectionRegistration);
 
         implementGetObservedType(observerCreator, observedType.getFieldDescriptor());
         if (observedQualifiers != null) {
@@ -345,7 +345,8 @@ public class ObserverGenerator extends AbstractGenerator {
 
     protected void createConstructor(ClassOutput classOutput, ClassCreator observerCreator, ObserverInfo observer,
             String baseName,
-            Map<InjectionPointInfo, String> injectionPointToProviderField, AnnotationLiteralProcessor annotationLiterals) {
+            Map<InjectionPointInfo, String> injectionPointToProviderField, AnnotationLiteralProcessor annotationLiterals,
+            ReflectionRegistration reflectionRegistration) {
 
         MethodCreator constructor;
         if (observer.isSynthetic()) {
@@ -381,9 +382,9 @@ public class ObserverGenerator extends AbstractGenerator {
                 if (builtinBean != null) {
                     builtinBean.getGenerator()
                             .generate(new GeneratorContext(classOutput, observer.getDeclaringBean().getDeployment(),
-                                    injectionPoint,
-                                    observerCreator, constructor, injectionPointToProviderField.get(injectionPoint),
-                                    annotationLiterals, observer));
+                                    injectionPoint, observerCreator, constructor,
+                                    injectionPointToProviderField.get(injectionPoint),
+                                    annotationLiterals, observer, reflectionRegistration));
                 } else {
                     if (injectionPoint.getResolvedBean().getAllInjectionPoints().stream()
                             .anyMatch(ip -> BuiltinBean.INJECTION_POINT.hasRawTypeDotName(ip.getRequiredType().name()))) {
@@ -396,7 +397,8 @@ public class ObserverGenerator extends AbstractGenerator {
                         ResultHandle annotationsHandle = BeanGenerator.collectAnnotations(classOutput, observerCreator,
                                 observer.getDeclaringBean().getDeployment(), constructor,
                                 injectionPoint, annotationLiterals);
-                        ResultHandle javaMemberHandle = BeanGenerator.getJavaMemberHandle(constructor, injectionPoint);
+                        ResultHandle javaMemberHandle = BeanGenerator.getJavaMemberHandle(constructor, injectionPoint,
+                                reflectionRegistration);
 
                         // Wrap the constructor arg in a Supplier so we can pass it to CurrentInjectionPointProvider c'tor.
                         ResultHandle delegateSupplier = constructor.newInstance(

--- a/integration-tests/main/src/main/java/io/quarkus/it/metrics/DummyGauge.java
+++ b/integration-tests/main/src/main/java/io/quarkus/it/metrics/DummyGauge.java
@@ -1,0 +1,30 @@
+package io.quarkus.it.metrics;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.interceptor.AroundConstruct;
+
+import org.eclipse.microprofile.metrics.MetricUnits;
+import org.eclipse.microprofile.metrics.annotation.Gauge;
+
+import io.quarkus.runtime.StartupEvent;
+
+/**
+ * The purpose of this bean is to test that ArC automatically registers the constructor for reflection if an
+ * {@link AroundConstruct} interceptors is used. We use a dedicated bean to make sure the class is not registered by another
+ * extension, ie. by resteasy in case of JAX-RS resource.
+ * 
+ * See https://github.com/quarkusio/quarkus/issues/6898
+ */
+@ApplicationScoped
+public class DummyGauge {
+
+    void init(@Observes StartupEvent event) {
+    }
+
+    @Gauge(name = "dummyGauge", unit = MetricUnits.NONE)
+    public Long dummyGauge() {
+        return 42l;
+    }
+
+}

--- a/integration-tests/main/src/main/java/io/quarkus/it/metrics/DummyGauge.java
+++ b/integration-tests/main/src/main/java/io/quarkus/it/metrics/DummyGauge.java
@@ -8,6 +8,7 @@ import org.eclipse.microprofile.metrics.MetricUnits;
 import org.eclipse.microprofile.metrics.annotation.Gauge;
 
 import io.quarkus.runtime.StartupEvent;
+import io.quarkus.runtime.annotations.RegisterForReflection;
 
 /**
  * The purpose of this bean is to test that ArC automatically registers the constructor for reflection if an
@@ -17,6 +18,7 @@ import io.quarkus.runtime.StartupEvent;
  * See https://github.com/quarkusio/quarkus/issues/6898
  */
 @ApplicationScoped
+@RegisterForReflection(fields = false)
 public class DummyGauge {
 
     void init(@Observes StartupEvent event) {

--- a/integration-tests/main/src/test/java/io/quarkus/it/main/MetricsTestCase.java
+++ b/integration-tests/main/src/test/java/io/quarkus/it/main/MetricsTestCase.java
@@ -160,6 +160,14 @@ public class MetricsTestCase {
                 .body(not(containsString("io_smallrye_metrics")));
     }
 
+    /**
+     * This is to verify that a Gauge works in native mode when its owning bean is annotated with RegisterForReflection.
+     */
+    @Test
+    public void testDummyGauge() {
+        assertMetricExactValue(metricsPrefix + "DummyGauge_dummyGauge", "42.0");
+    }
+
     private void assertMetricExactValue(String name, String val) {
         RestAssured.when().get("/metrics").then()
                 .body(containsString(name + " " + val));

